### PR TITLE
feat(output): auto-open HTML file after scan

### PR DESCRIPTION
**Pull request by hogo6002**
Added support to automatically open the HTML file in users' default browser after a scan finishes.

If the serve flag is enabled, the HTML file will be saved in a temporary directory and then served on localhost.

If only specified the `--format html`, open the generated HTML file by using the right `open` command for each OS (like `xdg-open` for Linux, `start` for Windows, and `open` for macOS). Simultaneously, the file is served on localhost for easy remote access.

### CODE REVIEW

**Comment by hogo6002** [comment]
Hmm, thinking a bit more, if we host it on a localhost port, users could potentially use SSH port forwarding to access the website if they are remotely accessing the server. But I don't know how likely this case would be.
`ssh -L local_port:destination_server_ip:remote_port ssh_server_hostname`

**Comment by oliverchang** [comment]
> Hmm, thinking a bit more, if we host it on a localhost port, users could potentially use SSH port forwarding to access the website if they are remotely accessing the server. But I don't know how likely this case would be. `ssh -L local_port:destination_server_ip:remote_port ssh_server_hostname`

This is the workflow I typically use whenever I need to inspect a web page generated by some tool (e.g. documentation)

**Comment by hogo6002** [comment]
> This is the workflow I typically use whenever I need to inspect a web page generated by some tool (e.g. documentation)

Updated code to also host the HTML file on localhost

File: cmd/osv-scanner/scan/main.go [hunk 1]
```diff
@@ -296,6 +300,54 @@ func action(context *cli.Context, stdout, stderr io.Writer) (reporter.Reporter,
...
+		r.Errorf("Failed to open: %s.\n Please manually open the outputted HTML file: %s\n", err, outputPath)
+	}
+
+	// Serve the single HTML file for remote accessing.
+	// The program will keep running to serve the HTML report on localhost
+	// until the user manually terminates it (e.g. using Ctrl+C).
+	// This could be done in a separate goroutine to avoid blocking the main thread,
+	// but since this is basically the last step in generating the HTML output,
+	// it's kept in the main thread for better maintainability and readability.
+	servePort := "8000"
```
[the remark this answers is not in the document]
Comment (hogo6002) [reply cmd/osv-scanner/scan/main.go:340]: I think this is primarily for user to remote access.

[the remark this answers is not in the document]
Comment (another-rex) [reply cmd/osv-scanner/scan/main.go:340]: I'm assuming open will most likely fail if it's a machine that is being sshed into, which is why I'm thinking this should start the server when xdg-open fails.

**Review by another-rex — COMMENTED** [review]
Let's add a flag `--host` or similar where we assume output is a temporary file, format is html, and run the hosting part of the code

File: cmd/osv-scanner/scan/main.go [hunk 2]
```diff
@@ -73,6 +78,10 @@ func Command(stdout, stderr io.Writer, r *reporter.Reporter) *cli.Command {
 					return fmt.Errorf("unsupported output format \"%s\" - must be one of: %s", s, strings.Join(reporter.Format(), ", "))
 				},
 			},
+			&cli.BoolFlag{
+				Name:  "host",
```
Comment (another-rex) [inline cmd/osv-scanner/scan/main.go:82]: nit: rename this to --serve which I think is the more common terminology

**Review by another-rex — APPROVED** [review]

**Pull request merged**